src/selectors/per-thread/composed.js

```diff
@@ -76,6 +82,28 @@ export function getComposedSelectorsPerThread(
       if (!jsTracerTable) {
         visibleTabs = visibleTabs.filter((tabSlug) => tabSlug !== 'js-tracer');
       }
+      let hasSamples = samples.length > 0 && stackTable.length > 0;
+      if (hasSamples) {
+        const stackIndex = ensureExists(samples.stack[0]);
+        if (stackTable.prefix[stackIndex] === null) {
+          // There's only a single stack frame, check if it's '(root)'.
+          const frameIndex = stackTable.frame[stackIndex];
+          const funcIndex = frameTable.func[frameIndex];
+          const stringIndex = funcTable.name[funcIndex];
+          if (stringTable.getString(stringIndex) === '(root)') {
+            // If the first sample's stack is only the root, check if any other
+            // sample is different.
+            hasSamples = samples.stack.some((s) => s !== stackIndex);
+          }
+        }
+      }
+
+      if (!hasSamples) {
+        visibleTabs = visibleTabs.filter(
+          (tabSlug) =>
+            !['calltree', 'flame-graph', 'stack-chart'].includes(tabSlug)
```

Comment:
Could you move this array to `app-logic/tab-handling.js` file so we can keep them in sync in case we add another tab that shows samples related visualization? Maybe we can name it something like `tabsShowingSampleData` or `sampleDataTabs` or something else?